src/modules/authentication/passport/oidc.strategy.ts

```diff
@@ -40,7 +50,11 @@ export class OpenIdConnectStrategy extends PassportStrategy(Strategy, 'oidc') {
             };
             return user;
         } catch (err: unknown) {
-            throw new UnauthorizedException();
+            if (err instanceof KeycloakUserNotFoundError) {
+                throw err;
+            } else {
+                throw new UnauthorizedException();
+            }
```

Comment:
Rethrowing errors is not pretty. Maybe only include the `client.userinfo` and `personRepo.findByKeycloakUserId` inside the try-catch and move the does-this-person-exist check outside.

Reply:
I agree that rethrowing the exception is not pretty but if this.personRepo.findByKeycloakUserId() throws an Exception it has to be mapped into an UnauthorizedException() otherwise an unknown Exception will be bubled up to the FE.